File: Neetcode 150/Two Pointer/two-sum-ii-input-array-is-sorted.cpp

// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        vector<int> ans;
        int size = numbers.size();
        if(size == 0) return ans;

        int left = 0, right = size-1;
        while(left < right){
            int currSum = numbers[left] + numbers[right];
            if(currSum == target){
                ans.push_back(left+1);
                ans.push_back(right+1);
                return ans;
            }else if(currSum > target) right--;
            else left++;
        }

        return ans;
    }
};

File: Neetcode-150-js/Two Pointers/two-sum-ii-input-array-is-sorted.js

// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let i = 0, j = numbers.length-1;
    while(i <= j){
        let currSum = numbers[i] + numbers[j];
        
        if(currSum === target) return [i+1, j+1];
        else if(currSum < target) i++;
        else j--;
    }
    return [];
};